modules/react/src/hooks.ts

```diff
@@ -53,43 +53,33 @@ export function useHubbleGl<ReactMapRef extends MapRef>({
     initialViewState
   );
 
-  const onMapLoad = useCallback(() => {
-    if (mapRef) {
-      const map = mapRef.current.getMap();
-      map.on('render', () => adapter.onAfterRender(nextFrame, map.areTilesLoaded()));
-    }
-  }, [adapter, nextFrame]);
-
-  if (!mapRef) {
+  const mapProps = useMemo(() => {
+    if (!mapRef) return {};
     return {
-      adapter,
-      cameraFrame,
-      setCameraFrame,
-      mapProps: {},
-      deckProps: adapter.getProps({
-        deck,
-        onNextFrame: nextFrame,
-        extraProps: {
-          layers
+      onLoad: () => {
+        if (mapRef.current) {
+          const map = mapRef.current.getMap();
+          map.on('render', () => adapter.onAfterRender(nextFrame, map.areTilesLoaded()));
         }
-      })
+      },
+      preventStyleDiffing: true
     };
-  }
+  }, [mapRef, adapter, nextFrame]);
+
+  const deckProps = useMemo(() => {
+    if (!deck) return undefined;
```

Comment:
```suggestion
    if (!deck) return {};
```
Should this be an empty object too so it can always be spread? `adapter.getProps` returns `{}` https://github.com/visgl/hubble.gl/blob/master/modules/core/src/adapters/deck-adapter.ts#L70